File: qc-domain/src/main/java/com/eager/questioncloud/domain/review/repository/QuestionReviewRepository.java
package com.eager.questioncloud.domain.review.repository;

import com.eager.questioncloud.common.PagingInformation;
import com.eager.questioncloud.domain.review.dto.QuestionReviewDetail;
import com.eager.questioncloud.domain.review.model.QuestionReview;
import java.util.List;

public interface QuestionReviewRepository {
    int getTotal(Long questionId);

    List<QuestionReviewDetail> getQuestionReviews(Long questionId, Long userId, PagingInformation pagingInformation);

    QuestionReview getMyQuestionReview(Long questionId, Long userId);

    QuestionReview findByIdAndUserId(Long reviewId, Long userId);

    Boolean isWritten(Long questionId, Long userId);

    QuestionReview save(QuestionReview questionReview);
}

File: qc-domain/qc-review/src/main/kotlin/com/eager/questioncloud/review/infrastructure/repository/QuestionReviewRepository.kt
package com.eager.questioncloud.review.infrastructure.repository

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.review.domain.QuestionReview
import com.eager.questioncloud.review.dto.ReviewerStatistics

interface QuestionReviewRepository {
    fun countByQuestionId(questionId: Long): Int

    fun findByQuestionIdWithPagination(questionId: Long, pagingInformation: PagingInformation): List<QuestionReview>

    fun findByQuestionIdAndUserId(questionId: Long, userId: Long): QuestionReview

    fun findByIdAndUserId(reviewId: Long, userId: Long): QuestionReview

    fun isWritten(userId: Long, questionId: Long): Boolean

    fun save(questionReview: QuestionReview): QuestionReview

    fun getReviewerStatistics(userIds: List<Long>): Map<Long, ReviewerStatistics>
}
